### PR TITLE
Add return statement support

### LINKED ISSUE #22215: Add return statement support
**Description:**

Add the `return-statement` (https://ballerina.io/ballerina-spec/spec.html#section_7.25) support to the new parser.

### PR DESCRIPTION
**Pull request by lochana-chathura**
Fixes #22215 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -19,12 +19,7 @@
 
 import io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.Action;
 import io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.Solution;
-import io.ballerinalang.compiler.internal.parser.tree.STCheckExpression;
-import io.ballerinalang.compiler.internal.parser.tree.STMissingToken;
-import io.ballerinalang.compiler.internal.parser.tree.STNode;
-import io.ballerinalang.compiler.internal.parser.tree.STNodeFactory;
-import io.ballerinalang.compiler.internal.parser.tree.STToken;
-import io.ballerinalang.compiler.internal.parser.tree.SyntaxKind;
+import io.ballerinalang.compiler.internal.parser.tree.*;
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:22]: Importing everything from a package is discouraged. We should import only the required classes.

Shall we revert this change?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 2]
```diff
@@ -160,6 +155,8 @@ public STNode resumeParsing(ParserRuleContext context, STNode... parsedNodes) {
                 return parseWhileKeyword();
             case BOOLEAN_LITERAL:
                 return parseBooleanLiteral();
+            case RETURN_KEYWORD:
+                return parseReturnStatement();
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:159]: Shouldn't this be `parseReturnKeyword()`?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 3]
```diff
@@ -2954,4 +2953,72 @@ private STNode parseCheckingKeyword() {
...
+     * @return Parsed node
+     */
+    private STNode parseReturnStatementRhs(SyntaxKind tokenKind,STNode returnKeyword) {
+        STNode expr;
+        STNode semicolon;
+
+        switch (tokenKind) {
+            case SEMICOLON_TOKEN:
+                expr = STNodeFactory.createEmptyNode();
+                semicolon = parseSemicolon();
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:3015]: Since `semicolon = parseSemicolon();` is common to both cases, we can take it out from the switch-case

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/STNodeFactory.java [hunk 4]
```diff
@@ -111,6 +111,13 @@ public static STNode createVariableDeclaration(SyntaxKind kind,
         return new STVariableDeclaration(kind, typeName, variableName, equalsToken, initializer, semicolonToken);
     }
 
+    public static STNode createReturnStatement(SyntaxKind kind,
+                                                   STNode returnKeyword,
+                                                   STNode actionOrExpr,
+                                                   STNode semicolonToken) {
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/STNodeFactory.java:117]: Indentation are not consistent

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/ReturnStatement.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        returnToken = createToken(0);
+        return returnToken;
+    }
+
+    public Node actionOrExpr() {
+        if (actionOrExpr != null) {
+            return actionOrExpr;
+        }
+
+        actionOrExpr = createToken(1);
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/ReturnStatement.java:45]: Since actionOrExpr is a `Node` and not a `Token`, we cannot use `createToken(1)` method.
Have a look at the `expression()` method in the `AssignmentStatement`

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java (hunk 1 above)
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:22]: sure

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java (hunk 2 above)
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:190]: ahh yeah. thanks

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/ReturnStatement.java (hunk 5 above)
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/ReturnStatement.java:45]: thanks. will check and correct

**Comment by lochana-chathura** [comment]
Made all requested changes. thanks
